### DISCUSSION TITLE
Separate Payment Entity from Order Entity

### DISCUSSION THREAD

**Discussion by victormattosvm** [woocommerce/woocommerce#35073] [WooCommerce HPOS]
Hello

I don't know if this topic has already been discussed.
It is very common to come across e-commerce platforms and marketplaces that separate the Payment entity from the Order entity, that is, the payment flow is different from the fulfillment flow.

Do you think about changing Woo's behavior?
 
Fulfillment standard flow (example):
- Request received
- Invoice issued (some cases)
- Order sent (or Awaiting pickup)
- Concluded

Payment standard flow (example):
- Invoice created
- Payment on hold
- Payment confirmed (or declined)

Because there is a different treatment, flexibility is one of the great benefits.

**Comment by barryhughes** [comment]
Hi @victormattosvm,

I agree there are several different flows, though I also think this sort of thing is already possible (albeit it may require some amount of custom coding to bring about).

Overall, this is probably out-of-scope for the [WooCommerce HPOS (High-Performance Order Storage) project](https://github.com/woocommerce/woocommerce/discussions/categories/woocommerce-hpos), however, and so it may be better to start by looking at current methods for achieving your goals and then, if appropriate, potentially framing an [enhancement request](https://github.com/woocommerce/woocommerce/issues/new?assignees=&labels=type%3A+enhancement%2Cstatus%3A+awaiting+triage&template=2-enhancement.yml&title=%5BEnhancement%5D%3A+) independently of the HPOS project.

Thanks!
